pkg/crc/preflight/preflight_checks_network_linux.go

```diff
@@ -0,0 +1,292 @@
+// +build linux
+
+package preflight
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/code-ready/crc/pkg/crc/systemd"
+	"github.com/code-ready/crc/pkg/crc/systemd/states"
+	crcos "github.com/code-ready/crc/pkg/os"
+)
+
+var nmPreflightChecks = [...]Check{
+	{
+		configKeySuffix:  "check-systemd-networkd-not-running",
+		checkDescription: "Checking if systemd-networkd is running",
+		check:            checkSystemdNetworkdIsNotRunning,
+		fixDescription:   "network configuration with systemd-networkd is not supported",
+		flags:            NoFix,
+	},
+	{
+		configKeySuffix:  "check-network-manager-installed",
+		checkDescription: "Checking if NetworkManager is installed",
+		check:            checkNetworkManagerInstalled,
+		fixDescription:   "NetworkManager is required and must be installed manually",
+		flags:            NoFix,
+	},
+	{
+		configKeySuffix:  "check-network-manager-running",
+		checkDescription: "Checking if NetworkManager service is running",
+		check:            checkNetworkManagerIsRunning,
+		fixDescription:   "NetworkManager is required. Please make sure it is installed and running manually",
+		flags:            NoFix,
+	},
+}
+
+var dnsmasqPreflightChecks = [...]Check{
+	{
+		configKeySuffix:    "check-network-manager-config",
+		checkDescription:   "Checking if /etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf exists",
+		check:              checkCrcNetworkManagerConfig,
+		fixDescription:     "Writing Network Manager config for crc",
+		fix:                fixCrcNetworkManagerConfig,
+		cleanupDescription: "Removing /etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf file",
+		cleanup:            removeCrcNetworkManagerConfig,
+	},
+	{
+		configKeySuffix:    "check-crc-dnsmasq-file",
+		checkDescription:   "Checking if /etc/NetworkManager/dnsmasq.d/crc.conf exists",
+		check:              checkCrcDnsmasqConfigFile,
+		fixDescription:     "Writing dnsmasq config for crc",
+		fix:                fixCrcDnsmasqConfigFile,
+		cleanupDescription: "Removing /etc/NetworkManager/dnsmasq.d/crc.conf file",
+		cleanup:            removeCrcDnsmasqConfigFile,
+	},
+}
+
+const (
+	crcDnsmasqConfigFile        = "crc.conf"
+	crcNetworkManagerConfigFile = "crc-nm-dnsmasq.conf"
+	crcNMDispatcherFile         = "99-crc.sh"
+)
+
+var (
+	crcDnsmasqConfigPath = filepath.Join(string(filepath.Separator), "etc", "NetworkManager", "dnsmasq.d", crcDnsmasqConfigFile)
```

Comment:
extract `NMConfigRoot` as `filepath.Join(string(filepath.Separator), "etc", "NetworkManager")` as this is beinng re-used serveral times

Reply:
Fixed.